### PR TITLE
Fix functional test (wrap micro nor cal test data into base docker image)

### PR DESCRIPTION
**Pull request by mark-idleman**
So, it turns out I [broke](https://github.com/replicahq/graphhopper/runs/3247477926) the functional test in #78 - even though I successfully force-ran the test by pasting it at the end of the test action that runs on every commit, the setup is slightly different in the real functional test action, which relies on the full base image generated by `Dockerfile.server` to actually contain the downloaded micro nor cal test files (which, with the setup I used in that PR, it does not).

The goal here: move the calling of `setup_tests.sh` into the docker build process itself, so that both the "intermediate" image (built with `Dockerfile.build`, which we run unit tests against) and the final server image contain all necessary micro nor cal test files. This will allow us to keep the functional test basically as-is, but it should actually work now :)

Tagged Kael as the main reviewer, but would love some eyes from @danielhfrank or @epotex or @JacobHayes

### CODE REVIEW

**Comment by mark-idleman** [comment]
Opening this up for review....I have the GH actions version working it seems (still need to test by building `Dockerfile.server` locally + running the functional test script), but I'm hitting issues with the Circle version (slack post with my best guess at issue [here](https://replicahq.slack.com/archives/C01GWQY6LTS/p1628209402002300)). Would love some insights if anyone has them! 

I'll post again + move out of draft once testing is done + the Circle version is working

File: .circleci/config.yml [hunk 1]
```diff
@@ -16,31 +16,29 @@ jobs:
       - run: &setup_gcloud
           name: Setup gcloud
           command: |
-            echo ${GCLOUD_SERVICE_KEY} | base64 -d > ${HOME}/gcp-key.json
-            gcloud auth activate-service-account --key-file ${HOME}/gcp-key.json
+            echo ${GCLOUD_SERVICE_KEY} | base64 -d > ${HOME}/repo/gcp-key.json
+            gcloud auth activate-service-account --key-file ${HOME}/repo/gcp-key.json
             gcloud --quiet config set project model-159019
+            pwd
```
Comment (mark-idleman) [inline .circleci/config.yml:22]: Printing some test info to help debug the issues in circle

**Comment by danielhfrank** [comment]
Seems reasonable in principle!

Sorry about the split CI thing - we'll talk in the meeting tomorrow, but hope to have that resolved very soon.

File: .circleci/config.yml (hunk 1 above)
Comment (kaelgreco) [reply .circleci/config.yml:22]: I assume we can take these out when we're confident it's working correctly (and all the echos and whatnot)

**Review by kaelgreco — COMMENTED** [review]
lgtm so far!

File: .circleci/config.yml (hunk 1 above)
Comment (mark-idleman) [reply .circleci/config.yml:22]: Yup! Just using them to try to understand why circle is unhappy 🙃

File: Dockerfile.build [hunk 2]
```diff
@@ -36,6 +36,22 @@ RUN npm install
...
+
+ARG MININORCAL_OSM_PATH
+ARG MININORCAL_GTFS_PATH
+ARG GOOGLE_CREDENTIALS_FILE
+RUN echo $GOOGLE_CREDENTIALS_FILE
+
+
+WORKDIR /graphhopper
+COPY . .
+RUN ls -la
```
Comment (epotex) [inline Dockerfile.build:50]: I'm assuming this is just for debug?

Comment (mark-idleman) [reply Dockerfile.build:50]: Yup 😀 https://github.com/replicahq/graphhopper/pull/81#discussion_r683875155

**Comment by mark-idleman** [comment]
Alright, #82 means that we don't need to worry about the circle issues anymore 😉 

Got this tested locally by building the `server-dev` image from the last `dev` image that was built, and then running the functional test script against it locally

**Comment by danielhfrank** [comment]
> Got this tested locally by building the server-dev image from the last dev image that was built, and then running the functional test script against it locally

Without getting involved in the details of this PR, reading this makes me think that it would be valuable to add the ability to run this action manually as well, in order to be able to make sure that changes don't break it. Doesn't have to be part of this PR though.

**Review by kaelgreco — APPROVED** [review]

**Comment by kaelgreco** [comment]
Going to merge since @mark-idleman is out!

**Pull request merged**